pkg/reconciler/configuration/configuration.go

```diff
@@ -235,13 +238,17 @@ func CheckNameAvailability(config *v1.Configuration, lister listers.RevisionList
 	// We only require spec equality because the rest is immutable and the user may have
 	// annotated or labeled the Revision (beyond what the Configuration might have).
 	if !equality.Semantic.DeepEqual(config.Spec.GetTemplate().Spec, rev.Spec) {
+		diff, err := kmp.SafeDiff(config.Spec.GetTemplate().Spec, rev.Spec)
+		if err != nil {
+			logger.Debugf("Revision's spec not equal to Configuration's spec template, diff: ", diff)
```

Comment:
Should we log the inequality (and the error) if there is an error here too for good measure? I have a hunch that this will fail if you have `resource` constraints in the configuration and... better safe than sorry.